python/ray/_raylet.pyx

```diff
@@ -898,16 +898,17 @@ cdef class CoreWorker:
 
         return RayObjectsToDataMetadataPairs(results)
 
-    def object_exists(self, ObjectRef object_ref):
+    def object_exists(self, ObjectRef object_ref, memory_store_only=False):
         cdef:
             c_bool has_object
+            c_bool is_in_plasma
             CObjectID c_object_id = object_ref.native()
 
         with nogil:
             check_status(CCoreWorkerProcess.GetCoreWorker().Contains(
-                c_object_id, &has_object))
+                c_object_id, &has_object, &is_in_plasma))
 
-        return has_object
+        return has_object and (not memory_store_only or not is_in_plasma)
```

Comment:
Augmenting `object_exists` with a flag to indicate whether or not to include Plasma existence was the easiest path to making this testable, and seemed like a tolerable addition given that (1) it's an internal API, (2) it preserves prior behavior by default, and (3) it could be useful for other Python tests.